pkg/pxc/encryption/keyring.go

```diff
@@ -0,0 +1,157 @@
+package encryption
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+const (
+	version    = "Keyring file version:2.0"
+	keyType    = "AES"
+	eof        = "EOF"
+	userID     = ""
+	keyLen     = 32
+	paddingLen = 5
+)
+
+func NewKeyring() ([]byte, error) {
+	key, err := key()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate new key: %v", err)
+	}
+
+	sha, err := shaSum(key)
+	if err != nil {
+		return nil, fmt.Errorf("failed to calculate sha256 sum")
+	}
+
+	keyring := bytes.NewBuffer(make([]byte, 0))
+	_, err = keyring.WriteString(version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write version: %v", err)
+	}
+
+	_, err = keyring.Write(key)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write key: %v", err)
+	}
+
+	_, err = keyring.WriteString(eof)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write eof: %v", err)
+	}
+
+	_, err = keyring.Write(sha)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write SHA sum : %v", err)
+	}
+
+	return keyring.Bytes(), nil
+}
+
+func shaSum(data []byte) ([]byte, error) {
+	h := sha256.New()
```

Comment:
I think you can just do `return sha256.Sum256(data)`

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/341/commits/4ad66bcdc1b1bced828cdf15a8d4b718130dd6da